test/src/vectorpostprocessors/TestParticleDataVectorPostprocessor.C

```diff
@@ -0,0 +1,60 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TestParticleDataVectorPostprocessor.h"
+#include "PICStudyBase.h"
+
+registerMooseObject("FenixTestApp", TestParticleDataVectorPostprocessor);
+
+InputParameters
+TestParticleDataVectorPostprocessor::validParams()
+{
+  InputParameters params = ParticleDataVectorPostprocessor::validParams();
+  return params;
+}
+
+TestParticleDataVectorPostprocessor::TestParticleDataVectorPostprocessor(
+    const InputParameters & parameters)
+  : ParticleDataVectorPostprocessor(parameters)
+{
+  _data_values.push_back(&declareVector("id"));
+}
+
+void
+TestParticleDataVectorPostprocessor::execute()
+{
+  // this does end up looping over the rays twice
+  // but since it will only be for testing this is fine
+  // helps to test the behaviour of the object people will actually
+  // use without requiring it to store meta data that is not important for them.
```

Comment:
```suggestion
  // This does end up looping over the rays twice,
  // but since it will only be for testing, this is fine.
  // It helps to test the behaviour of the object people will actually
  // use without requiring it to store meta data that is not important for them.
```